runtime/lib/backends/cuda/providers/default/math/batch_matmul.cc

```diff
@@ -71,87 +71,101 @@ BatchMatmulImpl<T>::BatchMatmulImpl(const OpAccessor &accessor) {
     const int lda = k;
     const int ldb = n;
     const int ldc = n;
-    BRT_CUBLAS_CHECK(cublasSgemmStridedBatched(
-        handle, CUBLAS_OP_N, CUBLAS_OP_N, n, m, k, &alpha, b_val, ldb,
-        batch_stride_B, a_val, lda, batch_stride_A, &beta, c_val, ldc,
-        batch_stride_C, batch_count));
+    BRT_CUBLAS_CHECK(cublasGemmStridedBatchedEx(
+        handle, CUBLAS_OP_N, CUBLAS_OP_N, n, m, k, &alpha, b_val, CUDA_R_32F,
+        ldb, batch_stride_B, a_val, CUDA_R_32F, lda, batch_stride_A, &beta,
+        c_val, CUDA_R_32F, ldc, batch_stride_C, batch_count, computeType,
+        CUBLAS_GEMM_DEFAULT));
   } else if (!lhs_transpose && rhs_transpose) {
     const int lda = k;
     const int ldb = k;
     const int ldc = n;
-    BRT_CUBLAS_CHECK(cublasSgemmStridedBatched(
-        handle, CUBLAS_OP_T, CUBLAS_OP_N, n, m, k, &alpha, b_val, ldb,
-        batch_stride_B, a_val, lda, batch_stride_A, &beta, c_val, ldc,
-        batch_stride_C, batch_count));
+    BRT_CUBLAS_CHECK(cublasGemmStridedBatchedEx(
+        handle, CUBLAS_OP_T, CUBLAS_OP_N, n, m, k, &alpha, b_val, CUDA_R_32F,
+        ldb, batch_stride_B, a_val, CUDA_R_32F, lda, batch_stride_A, &beta,
+        c_val, CUDA_R_32F, ldc, batch_stride_C, batch_count, computeType,
+        CUBLAS_GEMM_DEFAULT));
   } else if (lhs_transpose && !rhs_transpose) {
     const int lda = m;
     const int ldb = n;
     const int ldc = n;
-    BRT_CUBLAS_CHECK(cublasSgemmStridedBatched(
-        handle, CUBLAS_OP_N, CUBLAS_OP_T, n, m, k, &alpha, b_val, ldb,
-        batch_stride_B, a_val, lda, batch_stride_A, &beta, c_val, ldc,
-        batch_stride_C, batch_count));
+    BRT_CUBLAS_CHECK(cublasGemmStridedBatchedEx(
+        handle, CUBLAS_OP_N, CUBLAS_OP_T, n, m, k, &alpha, b_val, CUDA_R_32F,
+        ldb, batch_stride_B, a_val, CUDA_R_32F, lda, batch_stride_A, &beta,
+        c_val, CUDA_R_32F, ldc, batch_stride_C, batch_count, computeType,
+        CUBLAS_GEMM_DEFAULT));
   } else {
     const int lda = m;
     const int ldb = k;
     const int ldc = n;
-    BRT_CUBLAS_CHECK(cublasSgemmStridedBatched(
-        handle, CUBLAS_OP_T, CUBLAS_OP_T, n, m, k, &alpha, b_val, ldb,
-        batch_stride_B, a_val, lda, batch_stride_A, &beta, c_val, ldc,
-        batch_stride_C, batch_count));
+    BRT_CUBLAS_CHECK(cublasGemmStridedBatchedEx(
+        handle, CUBLAS_OP_T, CUBLAS_OP_T, n, m, k, &alpha, b_val, CUDA_R_32F,
+        ldb, batch_stride_B, a_val, CUDA_R_32F, lda, batch_stride_A, &beta,
+        c_val, CUDA_R_32F, ldc, batch_stride_C, batch_count, computeType,
+        CUBLAS_GEMM_DEFAULT));
   }
 }
 
 template <>
 void BatchMatmulImpl<__half>::Execute(const __half *a_val, const __half *b_val,
                                       __half *c_val, cublasHandle_t handle,
                                       cudaStream_t stream) {
-  const __half alpha = 1.0f;
-  const __half beta = 0.0f;
+  const float alpha = 1.0f, beta = 0.0f;
+  cublasComputeType_t computeType = CUBLAS_COMPUTE_32F;
```

Comment:
Should we use `CUBLAS_COMPUTE_16F` here?

Reply:
It's a `TODO` to support `compute_type = f16` here.